system/jlib/jptree.cpp

```diff
@@ -1010,18 +1010,18 @@ class SeriesPTIterator : implements IPropertyTreeIterator, public CInterface
 
 ///////////////////
 
-CPTValue::CPTValue(size32_t size, const void *data, bool binary, bool raw, CompressionMethod _compressType)
+CPTValue::CPTValue(size32_t size, const void *data, bool binary, bool raw, CompressionMethod currentCompressType, CompressionMethod preferredCompressType)
```

Comment:
currentCompreesType == COMPRESS_METHOD_NONE currently means (if !raw && binary), if over threshold use default compression method.

Perhaps should be COMPRESS_METHOD_DEFAULT, and allow COMPRESS_METHOD_NONE to be specified, which does nothing (aside from storing the type), to allow callers to explicitly say don't handle compression:
```
MemoryBuffer compressedMb; // my proprietary format, inc compression
ipt->setPropBin("blah", mb.length(), mb.toByteArray(), COMPRESS_METHOD_NONE);
```